### ISSUE TITLE
unusual warning in collections/mod.ts

### ISSUE THREAD

**Issue by timreichen** [denoland/deno_std#3412]
**Is your feature request related to a problem? Please describe.**
`collection/mod.ts` file logs a warning when imported:
https://github.com/denoland/deno_std/blob/fdabe55e2dd29df626ecb5b4496b091478298eaf/collections/mod.ts#L29-L33
I get that this is done to encourage single-export-file imports. But this is irritating, because it is the only `mod.ts` file that does this.
There are other `mod.ts` files (path/46kb, fs/74kb, yaml/92kb, toml/32kb, semver/33kb, etc.) that have far more bloat that collections/20kb.

**Describe the solution you'd like**
Remove the warning

**Describe alternatives you've considered**
- Remove the `mod.ts` files altogether
- Add warnings to all mod.ts files to be consistent

What do you think? Should I open a PR to remove the warning?

**Comment by karelklima** [comment]
My two cents: it is not the best developer experience in case you need import a lot of functions from collections in a single script. In my project I have ended up creating a local `collections.ts` file with the following content:

```typescript
export { groupBy } from "$std/collections/group_by.ts";
export { sortBy } from "$std/collections/sort_by.ts";
export { minOf } from "$std/collections/min_of.ts";
export { maxOf } from "$std/collections/max_of.ts";
export { filterEntries } from "$std/collections/filter_entries.ts";
export { distinct } from "$std/collections/distinct.ts";
export { chunk } from "$std/collections/chunk.ts";
export { mapValues } from "$std/collections/map_values.ts";
```

The above is not that developer friendly and hard to manage. Further, there is no way to learn about collections except to go to the manual / documentation and read through that. On the other hand if one could do `import * as collections from "$std/collections/mod.ts"`, you can have the collection functions autocompleted in IDE.

In the end it does not really matter whether or not to import stuff from STD through `mod.ts` files, but it should be consistent across all the submodules.

**Comment by iuioiua** [comment]
Another solution: remove the warning from that file only.

**Comment by kt3k** [comment]
I'm in favor of removing the warning from that file.

I believe we (including core team) are in agreement about that mod.ts and single-export files both have their own purposes/roles.

More context about this warning is available in: #2346 #2321

cc @ry What do you think?
